dataset_loaders/videos/movingMNIST.py

```diff
@@ -0,0 +1,222 @@
+            # Take a step along velocity.
+            y += v_y * y_rand  # + self.step_length_
+            x += v_x * x_rand  # + self.step_length_
+            # Bounce off edges.
+            for j in xrange(batch_size):
+                if x[j] <= 0:
+                    x[j] = 0
+                    v_x[j] = -v_x[j]
+                if x[j] >= 1.0:
+                    x[j] = 1.0
+                    v_x[j] = -v_x[j]
+                if y[j] <= 0:
+                    y[j] = 0
+                    v_y[j] = -v_y[j]
+                if y[j] >= 1.0:
+                    y[j] = 1.0
+                    v_y[j] = -v_y[j]
+            start_y[i, :] = y
+            start_x[i, :] = x
+
+        # Scale to the size of the canvas.
+        start_y = (canvas_size * start_y).astype(np.int32)
+        start_x = (canvas_size * start_x).astype(np.int32)
+        return start_y, start_x
+
+    def get_batch(self, verbose=False):
+        start_y, start_x = self.get_random_trajectory(
+            self.batch_size_ * self.num_digits_)
+
+        # minibatch data
+        # if self.background_ == 'zeros':
+        data = np.zeros((self.batch_size_, self.num_channels_,
+                         self.image_size_, self.image_size_,
+                         self.seq_length+1), dtype=np.float32)
+        # elif self.background_ == 'rand':
+        #     data = self._rng.rand(self.batch_size_, self.num_channels_,
+        #                          self.image_size_, self.image_size_,
+        #                          self.seq_length)
+
+        for j in xrange(self.batch_size_):
+            for n in xrange(self.num_digits_):
+
+                # get random digit from dataset
+                ind = self.indices_[self.row_]
+                self.row_ += 1
+                if self.row_ == self.data_.shape[0]:
+                    self._rng.shuffle(self.indices_)
+                    self.row_ = 0
+                digit_image = self.data_[ind, :, :]
+                zoom_factor = 1
+                if self.digits_sizes_ != 28:
+                    zoom_factor = int(self.digits_sizes_[n]/28)
+                    digit_image = zoom(digit_image, zoom_factor)
+                digit_size = digit_image.shape[0]
+
+                # if self.mode_ == 'squares':
+                #     digit_size = self._rng.randint(5, 20)
+                #     digit_image = np.ones((digit_size, digit_size),
+                #                           dtype=np.float32)
+
+                # generate video
+                for i in xrange(self.seq_length+1):
+                    top = start_y[i, j * self.num_digits_ + n]
+                    left = start_x[i, j * self.num_digits_ + n]
+                    bottom = top + digit_size
+                    right = left + digit_size
+                    data[j, :, top:bottom, left:right, i] = self.Overlap(
+                        data[j, :, top:bottom, left:right, i], digit_image)
+
+        data = np.transpose(data, (0, 4, 2, 3, 1))
```

Comment:
I suggest to create `data` with the correct shape from the beginning and to simply transpose the digits when you add them to `data`, rather than transposing `data` here.